handler/src/main/java/io/netty/handler/ssl/OpenSslSessionCache.java

```diff
@@ -145,28 +135,26 @@ public final boolean sessionCreated(long ssl, long sslSession) {
         if (engine == null) {
             return false;
         }
-
+        final OpenSslSession session = engine.sessionCreated(sslSession);
         synchronized (this) {
-            // Mimic what OpenSSL is duing and expunge every 255 new sessions
+            // Mimic what OpenSSL is doing and expunge every 255 new sessions
             // See https://www.openssl.org/docs/man1.0.2/man3/SSL_CTX_flush_sessions.html
             if (++sessionCounter == 255) {
                 sessionCounter = 0;
-                expungeInvalidSessions();
+                expungeInvalidSessions(System.currentTimeMillis());
             }
 
-            final OpenSslSession session;
-            try {
-                session = engine.sessionCreated(sslSession);
-            } catch (SSLException e) {
-                // TODO: Should we log this ?
+            if (session == null) {
                 return false;
             }
+            assert session.refCnt() >= 1;
             if (!sessionCreated(session)) {
                 return false;
```

Comment:
Should we release the `session` before returning `false` from here?

Reply:
no as we only call `retain()` once we put it in the cache a few lines below.